### ISSUE TITLE
🐛 BUG: Returning HTML element from if-else block causes error

### ISSUE THREAD

**Issue by jcha0713** [withastro/astro#3346]
### What version of `astro` are you using?

1.0.0-beta.20

### Are you using an SSR adapter? If so, which one?

None

### What package manager are you using?

npm

### What operating system are you using?

Mac

### Describe the Bug

When I try to return HTML elements from a if-else block in `.astro` file, the compiler gives me an error.

A simple example for this would be like this:
```astro
---
let isTrue = true;
---

// ...

if (isTrue) {
  return <p>it is true</p>
} else {
  return <p>it is not true</p>
}
```

I believe this should be able to return the element based on the condition, but it says it found an unexpected "return"

<img width="1360" alt="Screen Shot 2022-05-11 at 11 37 09" src="https://user-images.githubusercontent.com/29053796/167921915-cff864ce-1f4f-4a55-879c-f29be10da149.png">

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-t5dfh4?file=src/pages/index.astro

### Participation

- [X] I am willing to submit a pull request for this issue.

**Comment by natemoo-re** [comment]
Yep, this is expected. Astro files are not JSX, anything below the `---` fences should be markup, `return` is not allowed.

```astro
---
let isTrue = true;
---

<!-- ... -->
{isTrue ? <p>it is true</p> : <p>it is not true</p>}
```

**Comment by Chrissdroid** [comment]
Wait @natemoo-re, take a look at the stackblitz repro, even your function shortcut is broken

see: https://stackblitz.com/edit/github-t5dfh4-urcvqk
